cpp/include/cudf/types.hpp

```diff
@@ -327,6 +328,21 @@ inline bool operator!=(data_type const& lhs, data_type const& rhs) { return !(lh
  */
 std::size_t size_of(data_type t);
 
+/**
+ * @brief Combines two hashed values into a single hashed value.
+ *
+ * Adapted from boost hash_combine, modified for 32/64-bit
+ * https://stackoverflow.com/a/4948967/1550940
```

Comment:
We can't just use code off the internet without a proper license and SWIPAT. @fondaing

Reply:
added license